### PR TITLE
[INLONG-5909][Agent] Fix WatchService is no effect in the k8s

### LINKED ISSUE #5909: [Bug][Agent] WatchService is no effect in the k8s
### What happened

WatchService is no effect in the k8s when  write data.
To trigger a new task when Data changes. 

### What you expected to happen

 Don't trigger a new task when Data changes. 

### How to reproduce

No

### Environment

_No response_

### InLong version

master

### InLong Component

InLong Agent

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by GanfengTan**
Fix WatchService is no effect in the k8s.

- Fixes #5909 

### Motivation

Trigger a new task when Data changes, for example: add data.

### Modifications

Validate the job for same GroupId and StreamId.
### Verifying this change

*(Please pick either of the following options)*

- [x] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

### CODE REVIEW

**Review by yunqingmoswu — COMMENTED** [review]
Please handle the unit test error.

File: inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/db/JobProfileDb.java [hunk 1]
```diff
@@ -192,4 +197,20 @@ public List<JobProfile> getJobsByState(StateSearchKey stateSearchKey) {
         }
         return profileList;
     }
+
+    /**
+     * check local job state.
+     *
+     * @return list of job profile.
```
Comment (healchow) [inline inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/db/JobProfileDb.java:204]: The result type is a Map, please change the Java doc, and add the desc for the key-value.

File: inlong-agent/agent-core/src/main/java/org/apache/inlong/agent/core/job/JobManager.java [hunk 2]
```diff
@@ -241,6 +241,8 @@ public Runnable dbStorageCheckThread() {
             while (isRunnable()) {
                 try {
                     jobProfileDb.removeExpireJobs(jobDbCacheTime);
+                    Map<String, String> jobStateMap = jobProfileDb.getJobsState();
```
Comment (healchow) [inline inlong-agent/agent-core/src/main/java/org/apache/inlong/agent/core/job/JobManager.java:244]: Excuse me, why just print the state into the log, it seems not useful.

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/trigger/DirectoryTrigger.java [hunk 3]
```diff
@@ -128,13 +134,28 @@ private void registerAllSubDir(PathPattern entity,
...
                 queue.offer(copiedJobProfile);
             }
         }
     }
 
+    private boolean validateJobProfile(JobProfile jobProfile, Path path) {
+        String groupId = jobProfile.get(JOB_GROUP_ID);
+        String streamId = jobProfile.get(JOB_STREAM_ID);
+        String pathUri = path.toUri().getPath();
+        String key = groupId.concat(DELIMITER_HYPHEN).concat(streamId).concat(DELIMITER_HYPHEN).concat(pathUri);
```
Comment (healchow) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/trigger/DirectoryTrigger.java:151]: Adding more comments, please.

File: inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/db/JobProfileDb.java (hunk 1 above)
Comment (GanfengTan) [reply inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/db/JobProfileDb.java]: done

File: inlong-agent/agent-core/src/main/java/org/apache/inlong/agent/core/job/JobManager.java (hunk 2 above)
Comment (GanfengTan) [reply inlong-agent/agent-core/src/main/java/org/apache/inlong/agent/core/job/JobManager.java]: check local job, submit state to manager in the future.

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/TextFileReader.java [hunk 4]
```diff
@@ -79,6 +81,10 @@ public void getData() throws IOException {
                         }
                         resultLines.add(splitLines[i].trim());
                     }
+                    if (data.startsWith(splitStr, data.length() - splitStr.length() - 1)) {
```
Comment (healchow) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/TextFileReader.java:84]: It's a bit difficult to understand here, can you add some explanations or examples?

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/fetcher/ManagerFetcher.java [hunk 5]
```diff
@@ -300,6 +307,24 @@ private void dealWithFetchResult(TaskResult taskResult) {
...
+        int type = ManagerOpEnum.getOpType(newProfile.getInt(JOB_OP)).getType();
+        if (ManagerOpEnum.ACTIVE.getType() != type || ManagerOpEnum.ADD.getType() != type) {
+            return false;
+        }
+        JobProfileDb jobProfileDb = agentManager.getJobProfileDb();
+        List<JobProfile> jobsByState = jobProfileDb.getJobsByState(StateSearchKey.ACCEPTED);
+        jobsByState.addAll(jobProfileDb.getJobsByState(StateSearchKey.RUNNING));
+        AtomicBoolean jobIsRunning = new AtomicBoolean(false);
+        jobsByState.forEach(jobProfile -> {
+            if (jobProfile.get(JOB_ID).equals(newProfile.get(JOB_ID))) {
```
Comment (healchow) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/fetcher/ManagerFetcher.java:320]: Will `jobProfile.get(JOB_ID)` be null?
Wouldn't it be better to use `Objects.equals()` here?

File: inlong-agent/agent-core/src/main/java/org/apache/inlong/agent/core/trigger/TriggerManager.java [hunk 6]
```diff
@@ -156,6 +169,35 @@ private Runnable jobFetchThread() {
...
+        try {
+            jobWrapper = jobWrapperMap.get(profile.getInstanceId());
+        } catch (Exception exception) {
+            LOGGER.warn("jobWrapper is null");
+            return false;
+        }
+        List<Task> tasks = jobWrapper.getAllTasks();
+        for (Task task : tasks) {
+            JobProfile runJobProfile = task.getJobConf();
+            if (runJobProfile.get(JOB_DIR_FILTER_PATTERN).equals(profile.get(JOB_DIR_FILTER_PATTERN))) {
```
Comment (healchow) [inline inlong-agent/agent-core/src/main/java/org/apache/inlong/agent/core/trigger/TriggerManager.java:183]: Will `jobProfile.get(JOB_DIR_FILTER_PATTERN)` be null?
Wouldn't it be better to use `Objects.equals()` here?

File: inlong-agent/agent-core/src/main/java/org/apache/inlong/agent/core/job/JobWrapper.java [hunk 7]
```diff
@@ -169,4 +170,11 @@ public void addCallbacks() {
     public List<Task> getAllTasks() {
         return allTasks;
     }
+
```
Comment (healchow) [inline inlong-agent/agent-core/src/main/java/org/apache/inlong/agent/core/job/JobWrapper.java:173]: Suggest adding Java doc for public methods.

File: inlong-agent/agent-core/src/main/java/org/apache/inlong/agent/core/job/Job.java [hunk 8]
```diff
@@ -83,8 +88,21 @@ public void setJobInstanceId(String jobInstanceId) {
      * @return taskList
      */
     public List<Task> createTasks() {
+        return getTasks(this.jobConf);
+    }
+
+    /**
+     * build task from job config
+     *
+     * @param jobConf
```
Comment (healchow) [inline inlong-agent/agent-core/src/main/java/org/apache/inlong/agent/core/job/Job.java:97]: Please add a description for the param and return.

**Review by healchow — CHANGES_REQUESTED** [review]

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/fetcher/ManagerFetcher.java (hunk 5 above)
Comment (GanfengTan) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/fetcher/ManagerFetcher.java:320]: done

File: inlong-agent/agent-core/src/main/java/org/apache/inlong/agent/core/trigger/TriggerManager.java (hunk 6 above)
Comment (GanfengTan) [reply inlong-agent/agent-core/src/main/java/org/apache/inlong/agent/core/trigger/TriggerManager.java:183]: done

File: inlong-agent/agent-core/src/main/java/org/apache/inlong/agent/core/job/JobWrapper.java (hunk 7 above)
Comment (GanfengTan) [reply inlong-agent/agent-core/src/main/java/org/apache/inlong/agent/core/job/JobWrapper.java:173]: done

File: inlong-agent/agent-core/src/main/java/org/apache/inlong/agent/core/job/Job.java (hunk 8 above)
Comment (GanfengTan) [reply inlong-agent/agent-core/src/main/java/org/apache/inlong/agent/core/job/Job.java:97]: done

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/TextFileReader.java (hunk 4 above)
Comment (GanfengTan) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/TextFileReader.java:86]: done

**Review by ciscozhou — APPROVED** [review]

**Review by EMsnap — APPROVED** [review]

**Review by healchow — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**